cmd/cvc-operator/controller/cvc_controller.go

```diff
@@ -654,9 +677,44 @@ func (c *CVCController) deletePDBIfNotInUse(cvc *apis.CStorVolumeClaim) error {
 		err = apispdb.KubeClient().
 			WithNamespace(openebsNamespace).
 			Delete(pdbName, &metav1.DeleteOptions{})
+		if k8serror.IsNotFound(err) {
+			klog.Infof("pdb %s of volume %s was already deleted", pdbName, cvc.Name)
+			return nil
+		}
 		if err != nil {
 			return err
 		}
+		klog.Infof("Successfully deleted the PDB %s of volume %s", pdbName, cvc.Name)
+	}
+	return nil
+}
+
+// scaleVolumeReplicas identifies whether it is scaleup or scaledown case of
+// volume replicas. If user added entry of pool info under the spec then changes
+// are treated as scaleup case. If user removed poolInfo entry from spec then
+// changes are treated as scale down case. If user just modifies the pool entry
+// info under the spec then it is a kind of migration which is not yet supported
+func (c *CVCController) scaleVolumeReplicas(cvc *apis.CStorVolumeClaim) error {
+	var err error
+	if len(cvc.Spec.Policy.ReplicaPoolInfo) > len(cvc.Status.PoolInfo) {
```

Comment:
is this check sufficient? we need to make sure that - all the content in PoolInfo need to be part of ReplicaPoolInfo also, and then, few more are there in it.

Reply:
These kind checks should be a head before that is admission server.